### PR TITLE
NRG: Log healthy status when no longer falling behind

### PR DESCRIPTION
**Pull request by neilalexander**
This makes it clear in the logs that the falling behind condition cleared up.

Signed-off-by: Neil Twigg <neil@nats.io>

### CODE REVIEW

File: server/raft.go [hunk 1]
```diff
@@ -1269,8 +1270,18 @@ func (n *raft) isCurrent(includeForwardProgress bool) bool {
 		return false
 	}
 
+	// If we were previously logging about falling behind, also log when the problem
+	// was cleared.
+	clearBehindState := func() {
+		if n.hcbehind {
+			n.warn("Health check OK, no longer falling behind")
```
Comment (ripienaar) [inline server/raft.go:1277]: can we add details here about what asset it is? also on the falling behind log

Comment (neilalexander) [reply server/raft.go:1277]: The `n.warn` is already prefixing with the Raft ID & group, I don't think `*raft` actually knows the real name of the asset though.

Comment (ripienaar) [reply server/raft.go:1277]: ah of course, good (well unfortunately always logging the raft group names not the asset but ok its something)

File: server/raft.go [hunk 2]
```diff
@@ -1269,8 +1270,18 @@ func (n *raft) isCurrent(includeForwardProgress bool) bool {
...
+			n.hcbehind = false
+		}
+	}
+
 	// Make sure we are the leader or we know we have heard from the leader recently.
 	if n.State() == Leader {
+		clearBehindState()
```
Comment (derekcollison) [inline server/raft.go:1284]: This not only applicable to leaders no?

Comment (neilalexander) [reply server/raft.go:1284]: I put this here just to cover the case of if you became leader in that time, to ensure the behind state gets cleared.

Comment (derekcollison) [reply server/raft.go:1284]: Gotcha..

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**